helper.go

```diff
@@ -277,6 +271,99 @@ func arrayToKV(m map[string]RedisMessage, arr []RedisMessage, keys []string) map
 	return m
 }
 
+func clusterMGet(client Client, ctx context.Context, keys []string) (ret map[string]RedisMessage, err error) {
+	ret = make(map[string]RedisMessage, len(keys))
+	slotCount := make(map[uint16]int, len(keys)/2)
+	for _, key := range keys {
+		slotCount[slot(key)]++
+	}
+	cmds := mgetcmdsp.Get(0, len(slotCount))
+	defer mgetcmdsp.Put(cmds)
+	for s := range slotCount {
+		var builder any = client.B().Mget()
+		var first = true
+		for _, key := range keys {
+			if slot(key) == s {
+				if first {
+					builder = builder.(intl.Mget).Key(key)
+					first = false
+				} else {
+					builder = builder.(intl.MgetKey).Key(key)
+				}
+			}
+		}
+		cmds.s = append(cmds.s, builder.(intl.MgetKey).Build().Pin())
```

Comment:
Hi @SoulPancake,

Right now, you count the number of keys for each slot, but then discard them, and then iterate over the whole keys slice for each slot, which I think is super inefficient.

How about storing the resulting indexes of cmd.s to the map[uint16]int map first? As I said [previously](https://github.com/redis/rueidis/pull/908#issuecomment-3422802256). So that you can use `slot(key)` to locate the index of cmds.s and only iterate the keys slice twice in total.